open_schools_platform/student_management/student/services.py

```diff
@@ -11,5 +12,16 @@ def create_student_profile(name: str, age: int) -> StudentProfile:
     return student_profile
 
 
-def can_user_create_student_profile_check(family: Family, user: User) -> bool:
+def can_user_interact_with_student_profile_check(family: Family, user: User) -> bool:
     return user.parent_profile in family.parent_profiles.all()
+
+
+def update_student_profile(*, student_profile: StudentProfile, data) -> StudentProfile:
+    non_side_effect_fields = ['age', 'name']
+    filtered_data = {key: value for key, value in data.items() if value is not None}
```

Comment:
We need to "clean" dict that we pass in data parameter from None values. Because if we don't do it, it will cause exceptions.
For example, we provide {"name": provided value, "age" None}  dict as data parameter. Our function will try to set None value for age field in student profile model. But age field can't be None, so it will cause exception.

Reply:
Can we move this function in common/utils file? It can be useful in defferent places.